internal/http/services/owncloud/ocdav/tpc.go

```diff
@@ -0,0 +1,418 @@
+// +-----------------+       Data          +----------------+
+// 												^
+// 												|
+// 												| COPY
+// 												|
+// 										   +----------+
+// 										   |  Client  |
+// 										   +----------+
+
+// handleTPCPull performs a GET request on the remote site and upload it
+// the requested reva endpoint.
+func (s *svc) handleTPCPull(ctx context.Context, w http.ResponseWriter, r *http.Request, ns string) {
+	src := r.Header.Get("Source")
+	dst := path.Join(ns, r.URL.Path)
+	overwrite := r.Header.Get("Overwrite")
+	depth := r.Header.Get("Depth")
+	if depth == "" {
+		depth = "infinity"
+	}
+
+	sublog := appctx.GetLogger(ctx).With().Str("src", src).Str("dst", dst).Logger()
+	sublog.Debug().Str("overwrite", overwrite).Str("depth", depth).Msg("TPC Pull")
+
+	overwrite = strings.ToUpper(overwrite)
+	if overwrite == "" {
+		overwrite = "T"
+	}
+
+	if overwrite != "T" && overwrite != "F" {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	if depth != "infinity" && depth != "0" {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	// get Gateway client
+	client, err := s.getClient()
+	if err != nil {
+		sublog.Error().Err(err).Msg("error getting grpc client")
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	// check if destination exists
+	ref := &provider.Reference{Path: dst}
+	dstStatReq := &provider.StatRequest{Ref: ref}
+	dstStatRes, err := client.Stat(ctx, dstStatReq)
+	if err != nil {
+		sublog.Error().Err(err).Msg("error sending grpc stat request")
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	if dstStatRes.Status.Code != rpc.Code_CODE_OK && dstStatRes.Status.Code != rpc.Code_CODE_NOT_FOUND {
+		HandleErrorStatus(&sublog, w, dstStatRes.Status)
+		return
+	}
+
+	if overwrite == "F" {
+		sublog.Warn().Str("overwrite", overwrite).Msg("Destination already exists")
+		w.WriteHeader(http.StatusPreconditionFailed) // 412, see https://tools.ietf.org/html/rfc4918#section-9.8.5
+		return
+	}
+
+	w.WriteHeader(http.StatusAccepted)
+
+	err = s.performHTTPPull(ctx, client, r, w, depth == "infinity", ns)
+	if err != nil {
+		sublog.Error().Err(err).Str("depth", depth).Msg("error descending directory")
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	fmt.Fprintf(w, "success: Created")
+}
+
+func (s *svc) performHTTPPull(ctx context.Context, client gateway.GatewayAPIClient, r *http.Request, w http.ResponseWriter, recurse bool, ns string) error {
+	src := r.Header.Get("Source")
+	dst := path.Join(ns, r.URL.Path)
+	size := 1024
+	log := appctx.GetLogger(ctx)
+	log.Debug().Str("src", src).Str("dst", dst).Msg("Performing HTTP Pull")
+
+	// get upload url
+	uReq := &provider.InitiateFileUploadRequest{
+		Ref: &provider.Reference{Path: dst},
+		Opaque: &typespb.Opaque{
+			Map: map[string]*typespb.OpaqueEntry{
+				"Upload-Length": {
+					Decoder: "plain",
+					// TODO: handle case where size is not known in advance
```

Comment:
We have a `sizedeferred` parameter which can be passed. It's already handled in the file systems (https://github.com/cs3org/reva/blob/master/pkg/storage/utils/localfs/upload.go#L109). Just need to handle it in the storageprovider https://github.com/cs3org/reva/blob/master/internal/grpc/services/storageprovider/storageprovider.go#L457

Since you're using the simple upload protocol, the size does not need to be provided, so I'd prefer this approach rather than passing an arbitrary constant